webpack/assets/javascripts/react_app/components/SettingsTable/SettingsTable.js

```diff
@@ -1,18 +1,43 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import { Table } from '../common/table';
+import { Table, Thead, Tr, Th, Tbody, Td } from '@patternfly/react-table';
 
-import createSettingsTableSchema from './SettingsTableSchema';
+import { translate as __ } from '../../common/I18n';
+import SettingNameCell from './components/SettingName';
+import SettingValueCell from './components/SettingValueCell';
 
-const SettingsTable = ({ settings }) => (
-  <Table
-    key="settings-table"
-    columns={createSettingsTableSchema}
-    rows={settings}
-  />
+const TableHead = () => (
+  <Thead>
+    <Tr ouiaId="setting-table-heading-row">
+      <Th width={30}> {__('Name')} </Th>
+      <Th width={35}> {__('Value')} </Th>
+      <Th width={35}> {__('Description')} </Th>
+    </Tr>
+  </Thead>
 );
 
+const ViewRow = ({ settings }) =>
```

Comment:
Nitpick: this is showing multiple rows, not a singe one.
```suggestion
const ViewRows = ({ settings }) =>
```

Or you could have the loop in `SettingsTable` itself only of course and properly make this a single row. Up to you.

On a related note: what's a best practice here how many components you define? `TableHead` is only used once and the same for `ViewRow`. Would it be better to just inline them in `SettingsTable`? Perhaps more a question for @MariaAga. It's more a question of something I'd like to understand.

Reply:
Great question, for me it's more readable when it's separated to multiple components, so I prefer this approach, but I am not sure if we have some official approach.

Reply:
Inline is less readable, we split into components to everything will be named and understandable. This is also the reason we have a limit of 300 lines for js files